dialer.go

```diff
@@ -141,7 +141,7 @@ func (d *customDialer) writeWARCFromConnection(reqPipe, respPipe *io.PipeReader,
 	close(recordChan)
 
 	if err != nil {
-		// Note: at the moment these errors don't go anywhere because wrapConnection calls us as a goroutine
+		d.client.errorChannel <- err
 		return err
```

Comment:
I wonder if we should return anything at all. It might be better to just not return any variable, if it has no purpose.